lib/projects/cabal/default.nix

```diff
@@ -0,0 +1,47 @@
+# Useful for Haskell projects managed with Cabal.
+
+{ symlinkJoin, writeShellScriptBin }:
+let
+  id = "cabal";
+
+  lib = import ../lib { inherit writeShellScriptBin; };
+
+  make = { project, haskellPackages }:
+    let
+      makeCommand = lib.makeCommand project;
+
+      cabalInstallPkg = haskellPackages.cabal-install;
+
+      cabal = makeCommand {
+        name = "${id}-cabal";
+        script = ''
+          cd "${project.srcPath}"
+          ${cabalInstallPkg}/bin/cabal "$@"
+        '';
+      };
+
+      commands = {
+        inherit cabal;
+      };
+
+      combinedCommandsPackage = symlinkJoin {
+        name = "${id}-commands-${project.groupName}-${project.projectName}";
+        paths = builtins.map ({ package, ... }: package) (builtins.attrValues commands);
+      };
+    in
+    {
+      inherit commands combinedCommandsPackage;
+    };
+in
+{
+  inherit id make;
+  inherit (lib) defineProject;
+  shellHook = ghc: ''
+    export NIX_GHC="${ghc}/bin/ghc"
+    export NIX_GHCPKG="${ghc}/bin/ghc-pkg"
+    export NIX_GHC_DOCDIR="${ghc}/share/doc/ghc/html"
+    # Export the GHC lib dir to the environment
+    # so ghcide knows how to source package dependencies.
+    export NIX_GHC_LIBDIR="$(${ghc}/bin/ghc --print-libdir)"
+  '';
```

Comment:
We may not actually need this, might be a fragment of when I first implemented the `cabal` project type. Could be wrong though.

Reply:
I think we should leave the `shellHook` in `lib/projects/haskell/default.nix` as-is, because it's required to use the correct GHC. However, I don't think it's required for `cabal` projects.